cpp/include/cuspatial/experimental/linestring_distance.cuh

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace cuspatial {
+
+/**
+ * @copybrief cuspatial::pairwise_linestring_distance
+ *
+ * The shortest distance between two linestrings is defined as the shortest distance
+ * between all pairs of segments of the two linestrings. If any of the segments intersect,
+ * the distance is 0.
+ *
+ * @tparam Cart2dItA iterator type for point array of the first linestring of each pair. Must meet
+ * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam Cart2dItB iterator type for point array of the second linestring of each pair. Must meet
+ * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OffsetIterator iterator type for offset array. Must meet the requirements of
+ * [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OutputIt iterator type for output array. Must meet the requirements of
+ * [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam std::iterator_traits<Cart2dItA>::value_type value type of `Cart2dItA`, must be
```

Comment:
Why must it be? Why can't it be `vec_2d`?  And if it must be, then why is it a template parameter? I think you made a similar comment on one of my PRs.

Reply:
I was under the impression that this is a geometric operation and we can't use this function with geographic types - all the math computations - from cross products to point to segment distance are only well-defined for cartesian coordinates. They won't get proper results if user start using them on geographical coordinates. I had originally believed that this is why we introduce strong types for geometric/geographic types, to prevent user from calling the API with the wrong types. If they want to use it with geographic coordinates, they should perform a projection first.

Reply:
OK. It may not be well-defined for lonlat_2d, but there may be some other type derived from vec_2d for which it is well-defined...  But I am indeed unsure of this, so we can leave it only defined for cartesian_2d for now.